File: examples/functions/hello.js
#!/usr/bin/env node

function main() {
    // Get payload from environment variable
    const payloadStr = process.env.FUNCTION_PAYLOAD || '{}';
    
    let payload;
    try {
        payload = JSON.parse(payloadStr);
    } catch (e) {
        payload = {};
    }
    
    const name = payload.name || 'World';
    
    const response = {
        message: `Hello, ${name}!`
    };
    
    console.log(JSON.stringify(response));
}

main();

File: examples/functions/hello.py
#!/usr/bin/env python3
import json
import os

def main():
    # Get payload from environment variable
    payload_str = os.environ.get('FUNCTION_PAYLOAD', '{}')
    
    try:
        payload = json.loads(payload_str)
    except:
        payload = {}
    
    name = payload.get('name', 'World')
    
    response = {
        'message': f'Hello, {name}!'
    }
    
    print(json.dumps(response))

if __name__ == '__main__':
    main()
